src/xspi/qspi.rs

```diff
@@ -179,6 +229,14 @@ pub trait QspiExt {
 }
 
 impl Qspi<stm32::QUADSPI> {
+    pub fn change_bank(&mut self, bank: BankSelect) {
+        self.rb.cr.modify(|_, w| w.dfm().clear_bit());
+        match bank {
+            BankSelect::One => self.rb.cr.modify(|_, w| w.fsel().clear_bit()),
+            BankSelect::Two => self.rb.cr.modify(|_, w| w.fsel().set_bit()),
+        }
```

Comment:
Bits `FSEL` and `DFM` can only be modified when the peripheral is not busy. The following check should be used to protect against that

```rust
// Wait for the peripheral to indicate it is no longer busy.
while self.is_busy().is_err() {}
```

It's already used several times in `mod.rs`. Instead of blocking you could also return an error from this function. Then the user would need to handle the case that a transaction is ongoing when `change_bank` is called.

Reply:
Thank you for your review.    
I decided to return an error and let the user handle potential retry as I don't want to block the running code.

Let me know if the documentation comment that I added are enough or not.

Reply:
The documentation is good and it's enough.

The `change_bank_unchecked` can only be restricted by adding a second generic type parameter to the `Qspi` type. That would then be a breaking change and perhaps too much complexity for solving this problem.

It's not ideal that the `change_back_unchecked` method is always available, but it's not a major problem either. Overall I'd be happy to accept the PR in the current state if you are also?

Reply:
> The change_bank_unchecked can only be restricted by adding a second generic type parameter to the Qspi type. That would then be a breaking change and perhaps too much complexity for solving this problem.

A change that would not be API breaking would be to return a new Qspi type (`QspiSwitchable` for example) from `bank_switch`. This new type would have an inner `Qspi` exposed for any operation on classic Qspi and the function `change_bank_unchecked` so the normal `Qspi` won't have it.
However creating a type just for that might not be optimal.

> I'd be happy to accept the PR in the current state if you are also?

If you are happy with the MR, and if you think that it is not mandatory to do any change to make the `change_back_unchecked` usable only when it actually does something , please do!